### ISSUE TITLE
Add file argument to git_log()?

### ISSUE THREAD

**Issue by maelle** [r-lib/gert#2]
I'd like to be able to retrieve the commits related to a path, like what

```
git log <path_to_file>
```

and for deleted files

```
git log --all --full-history -- <path-to-file>
```

gives.

🙏 😸

**Comment by jdblischak** [comment]
@maelle We're working on adding a similar feature to `git2r::commits()` in https://github.com/ropensci/git2r/pull/372. Our strategy is to adapt the libgit2 example script [log.c](https://github.com/libgit2/libgit2/blob/master/examples/log.c). Unfortunately our current attempt still has some differences between the results returned by `git log <path_to_file>` that we are trying to debug. Since both git2r and gert wrap libgit2, a similar solution could also work for this package. It'd be great if both packages could have this feature, so any advice or help troubleshooting our current implementation would be much appreciated!

cc @pcarbo

**Comment by jdblischak** [comment]
@maelle A quick update. We fixed our implementation in https://github.com/ropensci/git2r/pull/372 and the master branch now has the feature to filter commits by a file path. Since it is based on libgit2, it should be possible to port the feature to gert.

**Comment by jdblischak** [comment]
I wanted to follow up on this Issue. I am really impressed with gert. I especially like the function `git_stat_files()`. This would be really useful for me. However, I really need the functionality requested in this Issue, i.e. to obtain the list of commits for a given file.

Is there any interest in implementing this functionality? I'm happy to help test and/or implement. Would you be open to a PR that adds an argument to `git_log()` that filters the returned commits by the file path?

**Comment by jeroen** [comment]
I think we'll have to rewrite git_log substantially to support this. You can send a PR, otherwise I will try to look into this over the upcoming weeks.

**Comment by pcarbo** [comment]
@jeroen John & I are looking into this. We agree that this would be a substantial rewrite of `git_log`, and it would most likely make more sense to make this a separate function. Would you be open to a new function, e.g., `git_log_modifying`, as long as it is suitable?

**Comment by jeroen** [comment]
Yes that sounds fine, or you could call it `git_revwalk()` because you are probably going to need to [libgit2 revwalk api](https://libgit2.org/libgit2/#HEAD/type/git_revwalk)?

**Comment by maelle** [comment]
I was looking for an issue related to adding a path to a folder to `git_log()`, I suppose it'd be covered by this open issue. :smile_cat:

**Comment by maelle** [comment]
Related https://github.com/libgit2/libgit2/discussions/6081

**Comment by maelle** [comment]
https://github.com/libgit2/libgit2/issues/3041

**Comment by jeroen** [comment]
That took nearly 8 years, thank you for your patience 👵
